Delete all unnecessary singular Math entries

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54470 Rename DefaultBackend to CompositeExplicitAutograd
* #54466 Rename Math to CompositeImplicitAutograd
* **#54436 Delete all unnecessary singular Math entries**
* #54427 s/StructuredNativeFunctions/NativeFunctionsGroup/
* #54426 Rejigger to use NativeFunctionsGroup even without structured: True
* #54419 Move compute_native_function_declaration to its own dest module
* #54397 Replace some tensor.device().is_cpu() calls with direct tensor.is_cpu()

An operator entry with no dispatch table implicitly generates a Math
entry, so you don't need to define one yourself.  I also added
some asserts in the codegen to fail on these cases.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27235381](https://our.internmc.facebook.com/intern/diff/D27235381)

ezyang committed:
Delete all unnecessary singular Math entries

An operator entry with no dispatch table implicitly generates a Math
entry, so you don't need to define one yourself.  I also added
some asserts in the codegen to fail on these cases.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ailzhang reviewed (approved):
LGTM! 
Btw since we're banning this unnecessary line, there's one more documentation to update :D https://github.com/pytorch/pytorch/blame/master/aten/src/ATen/native/README.md#L464

ezyang committed:
Update on "Delete all unnecessary singular Math entries"


An operator entry with no dispatch table implicitly generates a Math
entry, so you don't need to define one yourself.  I also added
some asserts in the codegen to fail on these cases.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27235381](https://our.internmc.facebook.com/intern/diff/D27235381)

[ghstack-poisoned]

ailzhang reviewed (approved):
Thanks!

ezyang committed:
Update on "Delete all unnecessary singular Math entries"


An operator entry with no dispatch table implicitly generates a Math
entry, so you don't need to define one yourself.  I also added
some asserts in the codegen to fail on these cases.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27235381](https://our.internmc.facebook.com/intern/diff/D27235381)

[ghstack-poisoned]
